api/v1alpha1/contentlibraryitemimportrequest_types.go

```diff
@@ -0,0 +1,169 @@
+type ContentLibraryItemImportRequestSpec struct {
+	// Source is the source of the import request which includes an external URL
+	// pointing to a VM image template.
+	// Source and Target will be immutable if the SourceValid and TargetValid conditions are true.
+	// +required
+	Source ContentLibraryItemImportRequestSource `json:"source"`
+
+	// Target is the target of the import request which includes the content library item
+	// information and a ContentLibrary resource.
+	// Source and Target will be immutable if the SourceValid and TargetValid conditions are true.
+	// +required
+	Target ContentLibraryItemImportRequestTarget `json:"target"`
+
+	// TTLSecondsAfterFinished is the time-to-live duration for how long this
+	// resource will be allowed to exist once the import operation
+	// completes. After the TTL expires, the resource will be automatically
+	// deleted without the user having to take any direct action.
+	// If this field is unset then the request resource will not be
+	// automatically deleted. If this field is set to zero then the request
+	// resource is eligible for deletion immediately after it finishes.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	TTLSecondsAfterFinished *int64 `json:"ttlSecondsAfterFinished,omitempty"`
+}
+
+// ContentLibraryItemFileUploadStatus indicates the upload status of files belonging to the template.
+type ContentLibraryItemFileUploadStatus struct {
+	// InProgress lists the names of files that are being imported into vSphere.
+	// +optional
+	InProgress []string `json:"inProgress,omitempty"`
+
+	// Completed lists the names of files that have been successfully imported into vSphere.
+	// +optional
+	Completed []string `json:"completed,omitempty"`
+
+	// Failed lists the names of files that failed to be imported into vSphere.
+	// +optional
+	Failed []string `json:"failed,omitempty"`
+}
+
+// ContentLibraryItemImportRequestStatus defines the observed state of a
+// ContentLibraryItemImportRequest.
+type ContentLibraryItemImportRequestStatus struct {
+	// ItemRef is the reference to the target ContentLibraryItem resource of the import request.
+	// If the ContentLibraryItemImportRequest is deleted when the import operation fails or before
+	// the Complete condition is set to true, the import operation will be cancelled in vSphere
+	// and the corresponding vSphere Content Library Item will be deleted.
+	// +optional
+	ItemRef *LocalObjectRef `json:"itemRef,omitempty"`
+
+	// CompletionTime represents time when the request was completed.
+	// The value of this field should be equal to the value of the
+	// LastTransitionTime for the status condition Type=Complete.
+	// +optional
+	CompletionTime metav1.Time `json:"completionTime,omitempty"`
+
+	// StartTime represents time when the request was acknowledged by the
+	// controller.
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+
+	// FileUpload indicates the upload status of files belonging to the template.
```

Comment:
FileUpload -> FileUploadStatus

Reply:
TFTR! This is updated in a later MR, please review if it looks good:
https://github.com/vmware-tanzu/image-registry-operator-api/pull/24/files#diff-874262c41e9b5f7eef360d32fa447a429ec46fa01d505dd2ecc123e172812fcdR108